src/config.js

```diff
@@ -0,0 +1,17 @@
+/**
+* @fileoverview Mixpanel source code, modified for DSP requirements
+*
+* Derived from code in https://github.com/mixpanel/mixpanel-js
+*/
+
+var Config = {
+  DEBUG: false,
+  LIB_VERSION: '0.0.1',
+  DSP_WEB_UI_ROOTS: [
+    // Fully-qualified domain names for production DSP web UIs
+    'terra.bio',
```

Comment:
This brings up an interesting point: Terra is deployed on several domains. The main one is `app.terra.bio`, but there are others, and the list changes somewhat frequently (several times a year, on average). Since we're relying on this to filter sensitive information, it seems like we can't afford for this to go out-of-date. There are also other places that want to consume this info, so there's been talk of centralizing this list somewhere, but it hasn't happened yet.

Perhaps this list should be given as a parameter to `getDefaultProperties`, so that it can be maintained/loaded from some central source?

Reply:
> Perhaps this list should be given as a parameter to getDefaultProperties, so that it can be maintained/loaded from some central source?

Done in 20edf62.  We might want to reconsider that parameter later (more below), but this seems helpful for now.

In addition to adding that parameter to `getDefaultProperties` as you suggested, I've also made `DSP_WEB_UI_ROOTS` more robust.  Before, it was list of strings.  Now, it's a list of regular expressions.  This enables more flexible [`match`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match)ing of the referring domain.

> there's been talk of centralizing this list somewhere

I like that idea.  Until that more central solution is available, I propose we use `config.js` here as the central list insofar as analytics is concerned.  If we rely on the new `webRoots` parameter just introduced into `getDefaultProperties`, any apps that aren't aware of a new prod domain will leak sensitive information.  (E.g. Single Cell Portal could be unaware of a Terra prod domain change, get an incoming link including a workspace name, and leak.)